.github/workflows/push.yml

```diff
@@ -0,0 +1,52 @@
+name: CI
+
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        model: [ "hydrafw" ]
+      fail-fast: true
+
+    steps:
+      - name: Update package cache
+        run: sudo  apt  update  -y
+
+      - name: Install packages
+        run: sudo  apt  install  -y  --no-install-recommends  --no-install-suggests  bash  coreutils  tar  bzip2  git  make  python3  python3-pip
+
+      - name: Install python modules
+        run: python3  -m pip  install  GitPython intelhex  --upgrade
+
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
```

Comment:
Does adding `fetch-tags: true` allows to run the Python script without errors ?

Reply:
[Nope](https://github.com/hydrabus/hydrafw/actions/runs/8456176252/job/23165330015#step:8:15). Maybe it's because what is built here is not from specific release tag but from main devel branch with commit not related to any specific release tag.

Reply:
That's a bug in the v4 checkout action. Currently afk but you should find details in the respective repo (including workarounds).

Reply:
> That's a bug in the v4 checkout action. Currently afk but you should find details in the respective repo (including workarounds).

Yea... that whole `v4` thing brought a lot of headache to a lot of people. For example, with `upload v4` update they disabled ability to upload multiple artifacts from multiple matrix'es into the zip with the same name.

---

Anyway, to whom it may concern and who is just curious: _it seems_ `fetch-depth: 0` helps to fetch tags and _some_ meta information but not _branches_ meta info (which is needed to count `22` commits) because `fetch-...` options don't change the fact that the tree itself ends up on github server in detached state right before the build.